libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -15,4 +15,8 @@ export class ReadingListComponent {
   removeFromReadingList(item) {
     this.store.dispatch(removeFromReadingList({ item }));
   }
+
+  public markAsFinished(item): void {
+    this.store.dispatch(markBookAsFinished({ item, finishedDate:  new Date().toISOString() }));
```

Comment:
So you are setting this value in the action after all! Why set it in all the other places too?

Reply:
I made changes at other places to use it from action